lighthouse-core/test/gather/gatherers/dobetterweb/response-compression-test.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/* eslint-env mocha */
+
+const ResponseCompression =
+    require('../../../../gather/gatherers/dobetterweb/response-compression');
+const assert = require('assert');
+
+let options;
+let optimizedResponses;
+const traceData = {
+  networkRecords: [
+    {
+      _url: 'http://google.com/index.js',
```

Comment:
it's a little unfair to make you do this @wardpeet since every other change using network records mixes these regularly, including [one landed yesterday](https://github.com/GoogleChrome/lighthouse/pull/1579/files#diff-fd190dc3127d17941b6683b1a8c4ece6R76) :P But if you're willing, that would be great to have this solved.

If it gets too involved feel free to push off to a followup PR so you can get this one in.